### PR TITLE
CI: CircleCI configuration

### PR DESCRIPTION
**Pull request by olleolleolle**
## Description

In order to improve build speed, switch to Circle CI.

## Todos

- [x] Test Metadata published to CircleCI as JUnit XML
- [x] Deploy to RubyGems on tagging
- [x] See if YAML aliases can be used for the shared Steps for Ruby versions which run the _same_ Steps
- [ ] Document the story
- [ ] After enabling on lostisland/faraday: Add a build status badge

# In other PR's

- [ ] JRuby excludes for things it can't `require` (e.g. `patron`) - this will go into its own PR

### CODE REVIEW

**Comment by olleolleolle** [comment]
Initial, set up on my fork: https://circleci.com/workflow-run/b0f33b48-be38-4977-b634-c2969133e04b

Error: I get a `Errno:EADDRNOTAVAIL` instead of the wrapped error

```
      expected Faraday::ConnectionFailed, got #<Errno::EADDRNOTAVAIL: Failed to open TCP connection to localhost:4 (Cannot assign requested address - connect(2) for “localhost” port 4)> with backtrace:
```

**Comment by olleolleolle** [comment]
One of the JRuby issues is that this is raised. Some value is not one of those things:

https://github.com/tmm1/http_parser.rb/blob/54b17ba8c7d8d20a16dfc65d1775241833219cf2/ext/ruby_http_parser/org/ruby_http_parser/RubyHttpParser.java#L495-L503

**Comment by olleolleolle** [comment]
Test Metadata does not get so nicely printed when it's "Rakefile"-started. `rspec` should be how we start the thing.

image

**Comment by headius** [comment]
@olleolleolle Hey just missed you on IRC, is there a question I can answer? Looks like you're moving alone here...

**Comment by headius** [comment]
FWIW we also have a Gitter room, if that fits your flow better... https://gitter.im/jruby/jruby

File: README.md [hunk 1]
```diff
@@ -6,7 +6,6 @@ LATEST STABLE RELEASE v0.15 IS IN BRANCH [0.1x](https://github.com/lostisland/fa
 # Faraday
 
 [![Gem Version](https://badge.fury.io/rb/faraday.svg)](https://rubygems.org/gems/faraday)
-[![Build Status](https://travis-ci.org/lostisland/faraday.svg)](https://travis-ci.org/lostisland/faraday)
```
Comment (iMacTia) [inline README.md]: Is there an equivalent badge for CircleCI?

Comment (olleolleolle) [reply README.md]: Yes, but it must be done in the configuration for the official repository.

**Pull request merged**